engine/common/follower/pending_tree/pending_tree.go

```diff
@@ -9,44 +9,17 @@ import (
 	"github.com/onflow/flow-go/module/mempool"
 )
 
-// CertifiedBlock holds a certified block, it consists of a block and a QC which proves validity of block (QC.BlockID = Block.ID())
-// This is used to compactly store and transport block and certifying QC in one structure.
-type CertifiedBlock struct {
-	Block *flow.Block
-	QC    *flow.QuorumCertificate
-}
-
-// ID returns unique identifier for the certified block
-// To avoid computation we use value from the QC
-func (b *CertifiedBlock) ID() flow.Identifier {
-	return b.QC.BlockID
-}
-
-// View returns view where the block was produced.
-func (b *CertifiedBlock) View() uint64 {
-	return b.QC.View
-}
-
-// Height returns height of the block.
-func (b *CertifiedBlock) Height() uint64 {
-	return b.Block.Header.Height
-}
-
 // PendingBlockVertex wraps a block proposal to implement forest.Vertex
 // so the proposal can be stored in forest.LevelledForest
 type PendingBlockVertex struct {
-	CertifiedBlock
+	flow.CertifiedBlock
 	connectedToFinalized bool
 }
 
 var _ forest.Vertex = (*PendingBlockVertex)(nil)
 
 // NewVertex creates new vertex while performing a sanity check of data correctness.
-func NewVertex(certifiedBlock CertifiedBlock, connectedToFinalized bool) (*PendingBlockVertex, error) {
-	if certifiedBlock.Block.Header.View != certifiedBlock.QC.View {
-		return nil, fmt.Errorf("missmatched block(%d) and QC(%d) view",
-			certifiedBlock.Block.Header.View, certifiedBlock.QC.View)
-	}
```

Comment:
I replaced this by a universal check in the constructor: https://github.com/onflow/flow-go/blob/19837298cd5996bca2c338fd2b7845e15c6d9ead/model/flow/block.go#L85-L100